llama-index-core/llama_index/core/indices/property_graph/transformations/dynamic_llm.py

```diff
@@ -34,24 +33,42 @@ def default_parse_dynamic_triplets(
     """
     triplets = []
 
-    # Regular expression to match the structure of each dictionary in the list
-    pattern = r"{'head': '(.*?)', 'head_type': '(.*?)', 'relation': '(.*?)', 'tail': '(.*?)', 'tail_type': '(.*?)'}"
-
-    # Find all matches in the output
-    matches = re.findall(pattern, llm_output)
-
-    for match in matches:
-        head, head_type, relation, tail, tail_type = match
-        head_node = EntityNode(name=head, label=head_type)
-        tail_node = EntityNode(name=tail, label=tail_type)
-        relation_node = Relation(
-            source_id=head_node.id, target_id=tail_node.id, label=relation
-        )
-        triplets.append((head_node, relation_node, tail_node))
-
+    try:
+        # Attempt to parse the output as JSON
+        data = json.loads(llm_output)
+        if isinstance(data, list):
+            for item in data:
+                head = item.get('head')
+                head_type = item.get('head_type')
+                relation = item.get('relation')
+                tail = item.get('tail')
+                tail_type = item.get('tail_type')
+
+                if head and head_type and relation and tail and tail_type:
+                    head_node = EntityNode(name=head, label=head_type)
+                    tail_node = EntityNode(name=tail, label=tail_type)
+                    relation_node = Relation(
+                        source_id=head_node.id, target_id=tail_node.id, label=relation
+                    )
+                    triplets.append((head_node, relation_node, tail_node))
+        else:
+    except json.JSONDecodeError:
```

Comment:
Seems like the code is complete? What is the `else` here, if its not a list?

Reply:
Hello, my apologies. Had some debug code in the else that I forgot to remove. I made the necessary changes and removed the useless "else" for both parsers. If JSON output parsing fails, we immediately fall back on the regex parsing anyways.

Reply:
great thanks!